src/components/TextField/TextField.tsx

```diff
@@ -251,6 +255,20 @@ class TextField extends React.PureComponent<CombinedProps, State> {
       </div>
     ) : null;
 
+    const clearButtonMarkup =
+      type === 'search' &&
+      clearButton !== false &&
+      !disabled &&
+      normalizedValue ? (
```

Comment:
This logic can be simplified a lot.

```tsx
      clearButton &&
```

This gets rid of the check for the type of the field. Just checking if `clearButton` is falsy will be sufficient. We'd still want to display the button even if the field is disabled. And `nomalizedValue` should always be truthy here, since it's set to an empty string if null.

Reply:
Based on Sarah's feedback, you'd also want to not render if `normalizedValue === ''`